### DISCUSSION TITLE
UX: Can't intermix arguments and options

### DISCUSSION THREAD

**Discussion by noncombatant** [FiloSottile/age#242] [UX reports]
I keep getting this:

```
~/age/cmd/age % ./age -d blorp --output snort
Error: too many arguments.
age accepts a single optional argument for the input file.
[ Did age not do what you expected? Could an error be more useful? Tell us: https://filippo.io/age/report ]
~/age/cmd/age % ./age -d --output snort blorp 
... success
```

One way to handle this would be to make the input an option (`-i`, `--input`), and not have any arguments at all. (But see bug #30 about using arguments for subcommands.) That would be maximally consistent and clear.

**Comment by FiloSottile** [comment]
Thanks for the report! This is a duplicate of #27, which really suggests we need to fix this. Unfortunately `-i` is taken by `--identity`, but I feel like we can make this better by accepting intermixed flags and args, or by detecting the issue and printing a better error.
